R/estimate_KM.R

```diff
@@ -184,6 +185,15 @@ estimate_KM <- function(
     }
   }
 
+  # add strata labels ----------------------------------------------------------
+  if (!is.null(strata)) {
+    survfit_object[["strata_lbls"]] <-
+      lapply(as.list(strata), function(x) attr(data[[x]], "label") %||% x) %>%
+      rlang::set_names(strata)
+  }
```

Comment:
else if (is.null(strata){
  survfit_object[["strata_lbls"]] <- "Overall"
  names(survfit_object[["strata_lbls"]]) <- "Overall"
}

Reply:
After some spot checks, I noticed we also have the required information in the names attribute of the strata element. I'm OK with adding that information into a new list element, but it should be done consistently (same logic for the attribute and the list element)

Reply:
> else if (is.null(strata){ survfit_object[["strata_lbls"]] <- "Overall" names(survfit_object[["strata_lbls"]]) <- "Overall" }

This is for the label, not the levels (which are already handled)

> After some spot checks, I noticed we also have the required information in the names attribute of the strata element. I'm OK with adding that information into a new list element, but it should be done consistently (same logic for the attribute and the list element)

That object has the raw levels of the stratifying variables, not the labels for a strata variable

Reply:
I think they can still be aligned correct?
We need to have "Overall" in the legend when no strata are specified

Reply:
> I think they can still be aligned correct? We need to have "Overall" in the legend when no strata are specified

Reply:
Can you put a comment to explain the differences between the list element of labels and the attribute above in the code?

Also explain briefly why you would not add "Overall" here.

Reply:
more detail added in the comments